dts/bindings/options/openthread,config.yaml

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  OpenThread configuration node.
+
+  Example usage:
+  options {
+    openthread{
+      compatible = "openthread,config";
+      diag-gpios = <&gpio0 0  GPIO_ACTIVE_HIGH>,
+                   <&gpio1 0  GPIO_ACTIVE_LOW>;
+    };
+  };
+
+compatible: "openthread,config"
+
+properties:
+  diag-gpios:
+    type: phandle-array
+    description: |
+      This enables access to diagnostic GPIO pins. Each field consists of
+      GPIO pin's configuration: controller's phandle, pin number and configuration flags.
+    required: true
```

Comment:
Does this have to be `required`? I imagine we could add properties in the future, which could all be optional.